lib/pages/main_page.dart

```diff
@@ -380,6 +374,92 @@ class MainPageState extends ConsumerState<MainPage> with TickerProviderStateMixi
     provider.save(newList);
   }
 
+  Future<void> initPlatformState() async {
+    final handler = ShareHandler.instance;
+    try {
+      media = await handler.getInitialSharedMedia();
+
+      handler.sharedMediaStream.listen((SharedMedia media) {
+        if (!mounted) return;
+        handleMediaFromShare(media);
+      });
+      if (!mounted) return;
+
+      handleMediaFromShare(media);
+    } on PlatformException {}
+  }
+
+  void handleMediaFromShare(SharedMedia? media) {
+    if (media == null || media.content == null) return;
+
+    var parsed = Uri.tryParse(media.content!);
+
+    if (parsed == null) return;
+
+    importRecipeFromSharedUri(parsed);
+  }
+
+  Future<void> importRecipeFromSharedUri(Uri uri) async {
+    var url = uri.toString();
+    var result = await ShoppingListSync.checkRecipeInput(url, context);
+
+    if (result.body != "true") {
+      showInSnackBar(NSSLStrings.of(context).recipeCreateError());
+      return;
+    }
+
+    var currList = ref.read(currentListProvider);
```

Comment:
```suggestion
    var currList = ref.watch(currentListProvider);
```
Normally read shouldn't be used that often and you use watch in the next line, so no reason to not use watch here as well